ls1intum/Artemis
# `Development`: Make copying feedback with attached long feedback text more robust

### Checklist
#### General


#### Server


### Motivation and Context
For manual assessments the feedbacks of a submission get copied into a new one. This did not always work for long feedback texts attached to feedbacks probably due to some lazy fetching being too lazy.

### Description
During the copy process now explicitly fetches long feedback texts if the lazy fetching did not do so automatically.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Instructor
- 1 Programming Exercise with manual assessment enabled

1. Add a programming exercise test case that produces a long feedback message, e.g. by adding a `fail("abc".repeat(10_000));` inside the test.
2. As the student: Submit to the exercise.
3. As instructor: Start the manual assessment.
4. The long feedback for the previously defined test case should be loaded in the feedback details view when clicking on the result string (_x%, y of z passed_).
   - The long feedback test case should end with `[...] ...` when folded.
   - When unfolding with the `> ` symbol it should show the full long feedback or a download link from which you can download the full text.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| FeedbackService.java | 100% | ✅                           |
